--- pipeline.rb 2017-12-07 23:05:59.000000000 +0000
+++ pipeline_new.rb     2019-05-08 20:35:04.864638554 +0000
@@ -371,7 +371,9 @@
       return @ci_yaml_file if defined?(@ci_yaml_file)
 
       @ci_yaml_file =
-        if auto_devops_source?
+        if ci_yaml_file_path =~ URI::regexp # if this is a URL
+          open(ci_yaml_file_path).read() # open and read the contents
+        elsif auto_devops_source?
           implied_ci_yaml_file
         else
           ci_yaml_from_repo
